augraphy/augmentations/quasicrystal.py

```diff
@@ -0,0 +1,118 @@
+"""
+version: 0.0.1
+*********************************
+
+Dependencies:
+- PIL
+- opencv
+- numpy
+
+*********************************
+
+References:
+
+- Numpy Documentation: https://numpy.org/doc/
+
+- PIL Documentation: https://pillow.readthedocs.io/en/stable/
+
+- OpenCV Documentation: https://docs.opencv.org/4.x/
+
+- Quasicrystals Inspiration: http://mainisusuallyafunction.blogspot.com/2011/10/quasicrystals-as-sums-of-waves-in-plane.html
+
+*********************************
+
+
+
+"""
+import math
+import os
+import random
+import warnings
+
+import cv2
+import numpy as np
+from PIL import Image
+
+from augraphy.base.augmentation import Augmentation
+from augraphy.utilities.slidingwindow import PatternMaker
+
+warnings.filterwarnings("ignore")
+
+
+class PatternGenerator(Augmentation):
+    def __init__(self, imgx=512, imgy=512, n_rotation=(10, 15), p=1.0):
+        """
+        In this implementation we take a geometric plane and every point in the plane is shaded according
+        to its position,(x,y) coordinate. We take the pattern and perform a bitwise not operation so that it can
+        be added as an background to an image.This code is a python implementation of a QuasiPattern Distortion augmentation techniques
+        using PIL and the OpenCV libraries. This augmentation creates a new pattern image and superimposes it onto an input image.
+
+        To make the pattern more prominent
+        a. Increase the 'frequency' parameter: Increasing the frequency of the pattern will the it tightly populated and more prominent.
+        b. Decrease the 'n_rotation' parameter: Decreasing the number of rotations will make the pattern less symmetrical.
+
+        :param imgx: width of the pattern image. default is 512
+        :type imgx: int
+        :param imgy: height of the pattern image, default is 512
+        :type imgy: int
+        :param n_rotation: is the number of rotations applied to the pattern, default value lies
+                           between 10 and 15.
+        :type n_rotation: tuple (int)
+
+
+        """
+        super().__init__(p=p)
+        self.imgx = imgx  # width of the image
+        self.imgy = imgy  # hieght of the image
+        self.n_rotation = random.randint(
```

Comment:
Similar with the issue i mentioned above, so this should be initialized in the `__call__`.